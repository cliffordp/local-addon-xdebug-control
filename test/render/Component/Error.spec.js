const React = require( 'react' )
const Component = require( 'react' ).Component
const renderer = require( 'react-test-renderer' )
const colors = require( './../../../src/UI/colors' )

const context = {
	'React': React,
}

const Error = require( './../../../src/Component/Error' )( context )

describe( '<Error/>', function () {
	const defaultProps = {
		message: 'Woops',
		openIssueLink: 'http://theaveragedev.com',
		cta: 'Click here',
		color: colors.green(),
	}

	it( 'renders correctly with props', function () {
		const component = renderer.create( <Error {...defaultProps}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )
} )
