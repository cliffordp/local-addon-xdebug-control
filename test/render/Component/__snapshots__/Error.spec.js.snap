// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error/> renders correctly with props 1`] = `
<section
  className="Error"
  style={
    Object {
      "border": "2px solid #1FC37D",
      "color": "#1FC37D",
      "padding": ".25em",
    }
  }
>
  <div
    className="Error__Message"
  >
    <div
      style={
        Object {
          "fontSize": "120%",
          "marginBottom": "1em",
          "textAlign": "center",
        }
      }
    >
      Error generated in 
      <span
        className="Error__Message__Source"
      >
        Unknown
      </span>
    </div>
    <div
      className="Error__Message__Text"
    >
      Woops
    </div>
    <div
      className="Error__Message__CTA"
      style={
        Object {
          "fontSize": "120%",
          "marginTop": "1em",
          "textAlign": "center",
        }
      }
    >
      Click here
    </div>
  </div>
</section>
`;
